Clarify github issues vs forum posts in docs. (#13612)
ethereum/solidity: docs/contributing.rst

Reducing the source code that caused the issue to a bare minimum is always
very helpful, and sometimes even clarifies a misunderstanding.

> In case you want to start a language design discussion, a post in the
> `Solidity forum <https://forum.soliditylang.org/>`_ is preferable to
> a github issue (see :ref:`solidity_language_design`).

Reviewer: ```suggestion
For technical discussions about language design, a post in the
`Solidity forum <https://forum.soliditylang.org/>`_ is the correct place (see :ref:`solidity_language_design`).
```

> 

Reviewer 2: Just wanted to note that @NunoFilipeSantos recently suggested that this whole section should just be replaced with a link to the bug tracker and the text should simply be a part of the "new issue" template.

And in the template chooser we do have an option encouraging people to open forum discussions (added by @nikola-matic recently):
![solidity-issue-template-chooser-forum-discussion](https://user-images.githubusercontent.com/137030/194061502-7ebb23a1-527c-4fe9-9d8f-775ca4ede9f6.png)

Reviewer 2: On the other hand, even if we just include a link, having this warning in multiple places might still be a good idea to draw people's attention. We could even make it a warning here.

Author: Ah, nice, that all sounds great - I'm also fine with dropping this in favour of those then.
